joern-cli/frontends/rubysrc2cpg/src/test/scala/io/joern/rubysrc2cpg/querying/ProcParameterAndYieldTests.scala

```diff
@@ -1,84 +1,96 @@
-          )
-          cpg.method("bar").call.code("yield").astChildren.isCall.code("<proc-param-1>").name.l shouldBe List(
-            "<proc-param-1>"
-          )
-        }
-
-        "add a block argument" in {
-          val List(param1) = cpg.method("foo").parameter.code("&.*").l
-          param1.name shouldBe "<proc-param-0>"
-          param1.index shouldBe 1
-
-          val List(param2) = cpg.method("bar").parameter.code("&.*").l
-          param2.name shouldBe "<proc-param-1>"
-          param2.index shouldBe 1
-        }
-      }
-
-      "with yield arguments" should {
-        val cpg = code("def foo(x) yield(x) end")
-        "replace the yield with a call to the block parameter with arguments" in {
-          val List(call) = cpg.call.codeExact("yield(x)").astChildren.isCall.codeExact("<proc-param-0>").l
-          call.name shouldBe "<proc-param-0>"
-          call.argument.code.l shouldBe List("self", "x")
-        }
-
-      }
+
+  "a method with an explicit proc parameter should create an invocation of it's `call` member" in {
+    val cpg = code("def foo(&b) yield end")
+
+    val foo = cpg.method("foo").head
+
+    val bParam = foo.parameter.last
+    bParam.name shouldBe "b"
+    bParam.code shouldBe "&b"
+    bParam.index shouldBe 1
+
+    inside(foo.call.nameExact("call").argument.l) { case selfBase :: Nil =>
+      selfBase.code shouldBe "b"
     }
+  }
+
+  "a singleton method with an explicit proc parameter should create an invocation of it's `call` member" in {
+    val cpg = code("def self.foo(&b) yield end")
+
+    val foo = cpg.method("foo").head
 
-    "that don't have a yield nor a proc parameter" should {
-      val cpg1 = code("def foo() end")
-      val cpg2 = code("def self.foo() end")
-      val cpgs = List(cpg1, cpg2)
+    val bParam = foo.parameter.last
+    bParam.name shouldBe "b"
+    bParam.code shouldBe "&b"
+    bParam.index shouldBe 1
 
-      "not add a block argument" in {
-        forAll(cpgs)(_.method("foo").parameter.code("&.*").name.l should be(empty))
-      }
+    inside(foo.call.nameExact("call").argument.l) { case selfBase :: Nil =>
+      selfBase.code shouldBe "b"
     }
+  }
+
+  "a method with an implicit proc parameter should create an invocation using a unique parameter name" in {
+    val cpg = code("""
+        |def foo() yield end
+        |def self.bar() yield end
+        |""".stripMargin)
+
+    val foo = cpg.method("foo").head
+    val bar = cpg.method("bar").head
+
+    val fooParam = foo.parameter.last
+    fooParam.name shouldBe "<proc-param-0>"
+    fooParam.code shouldBe "&<proc-param-0>"
+    fooParam.index shouldBe 1
+
+    val barParam = bar.parameter.last
+    barParam.name shouldBe "<proc-param-1>"
+    barParam.code shouldBe "&<proc-param-1>"
+    barParam.index shouldBe 1
+
+    foo.call.nameExact("call").argument.isIdentifier.name.l shouldBe List("<proc-param-0>")
+    bar.call.nameExact("call").argument.isIdentifier.name.l shouldBe List("<proc-param-1>")
```

Comment:
the block/proc parameter is not passed as an argument into the block/proc itself. It is just the receiver. Also applies to your other tests.

Reply:
I was wrong about this. Lets keep it like it is.